graylog2-server/src/main/java/org/graylog2/inputs/transports/AbstractHttpTransport.java

```diff
@@ -119,49 +125,67 @@ public AbstractHttpTransport(Configuration configuration,
     }
 
     @Override
-    protected LinkedHashMap<String, Callable<? extends ChannelHandler>> getCustomChildChannelHandlers(MessageInput input) {
+    protected LinkedHashMap<String, Callable<? extends ChannelHandler>> getCustomChildChannelHandlers(
+            MessageInput input) {
         final LinkedHashMap<String, Callable<? extends ChannelHandler>> handlers = new LinkedHashMap<>();
         if (idleWriterTimeout > 0) {
             // Install read timeout handler to close idle connections after a timeout.
-            // This avoids dangling HTTP connections when the HTTP client does not close the connection properly.
-            // For details see: https://github.com/Graylog2/graylog2-server/issues/3223#issuecomment-270350500
-            handlers.put("read-timeout-handler", () -> new ReadTimeoutHandler(idleWriterTimeout, TimeUnit.SECONDS));
+            // This avoids dangling HTTP connections when the HTTP client does not close the
+            // connection properly.
+            // For details see:
+            // https://github.com/Graylog2/graylog2-server/issues/3223#issuecomment-270350500
+            handlers.put("read-timeout-handler",
+                    () -> new ReadTimeoutHandler(idleWriterTimeout, TimeUnit.SECONDS));
         }
 
-        handlers.put("decoder", () -> new HttpRequestDecoder(DEFAULT_MAX_INITIAL_LINE_LENGTH, DEFAULT_MAX_HEADER_SIZE, maxChunkSize));
+        handlers.put("decoder", () -> new HttpRequestDecoder(DEFAULT_MAX_INITIAL_LINE_LENGTH,
+                DEFAULT_MAX_HEADER_SIZE, maxChunkSize));
         handlers.put("decompressor", HttpContentDecompressor::new);
         handlers.put("encoder", HttpResponseEncoder::new);
         handlers.put("aggregator", () -> new HttpObjectAggregator(maxChunkSize));
-        handlers.put("http-forwarded-for-handler", () -> new HttpForwardedForHandler(enableForwardedFor, enableRealIpHeader, realIpHeaders, requireTrustedProxies, trustedProxies));
-        handlers.put("http-handler", () -> new HttpHandler(enableCors, authorizationHeader, authorizationHeaderValue, path));
+        handlers.put("http-forwarded-for-handler", () -> new HttpForwardedForHandler(enableForwardedFor,
+                enableRealIpHeader, realIpHeaders, requireTrustedProxies, trustedProxies));
+        handlers.put("http-handler",
+                () -> new HttpHandler(enableCors, authorizationHeader, authorizationHeaderValue, path));
         if (enableBulkReceiving) {
             handlers.put("http-bulk-newline-decoder",
-                    () -> new LenientDelimiterBasedFrameDecoder(maxChunkSize, Delimiters.lineDelimiter()));
+                    () -> new LenientDelimiterBasedFrameDecoder(maxChunkSize,
+                            Delimiters.lineDelimiter()));
+        }
+        if (enableJsonArrayProcessing) {
+            handlers.put("http-json-array-decoder",
+                    () -> new JsonArrayFrameDecoder(maxChunkSize));
```

Comment:
It feels like this is just another way to enable bulk receiving yet we are adding a second config option to enable it. What do you guys think about just removing the config option and adding it to the existing handlers if `enableBulkReceiving` is true? Currently, if you check both checkboxes things seem to work fine. @danotorrey @ryan-carroll-graylog 

```suggestion
                            Delimiters.lineDelimiter()));
                            handlers.put("http-json-array-decoder",
                                    () -> new JsonArrayFrameDecoder(maxChunkSize));
```

Reply:
@kingzacko1 I agree with two bulk options being problematic. I had the same thought. It feels like this would fit best with a Content Type select list like we have for some newer inputs, but that would require a migration and add complexity to the changes required.

I also like the idea of a single option (with both decoders) - definitely the most simple way to go. But, we might run into issues with formatted JSON with line breaks (or even log messages with line breaks) being split in an unexpected way. We could test to be certain. 

Looping in @Graylog2/architecture for input as well.